File: backend/models/feedback.js
const mongoose = require('mongoose');

const FeedbackSchema = new mongoose.Schema({
  userId: { type: String, required: true },
  message: { type: String, required: true },
  timestamp: { type: Date, required: true },
});

const Feedback = mongoose.model('Feedback', FeedbackSchema);
module.exports = Feedback;

File: frontend/lib/models/feedback.dart
class UserFeedback {
  final String userId; // ID of the user providing feedback
  final String message; // Feedback message
  final DateTime timestamp; // When the feedback was given

  UserFeedback({required this.userId, required this.message, required this.timestamp});

  // Convert a Feedback object to a Map
  Map<String, dynamic> toMap() {
    return {
      'userId': userId,
      'message': message,
      'timestamp': timestamp.toIso8601String(),
    };
  }
}
